modules/nf-core/ctree/templates/main_script.R

```diff
@@ -0,0 +1,198 @@
+    parsed_args[! is.na(parsed_args)]
+}
+
+opt = list(
+    prefix = ifelse('$task.ext.prefix' == 'null', '$meta.id', '$task.ext.prefix')
+)
+args_opt = parse_args('$task.ext.args')
+for ( ao in names(args_opt)) opt[[ao]] = args_opt[[ao]]
+
+
+# Load packages
+
+library(ctree)
+library(mobster)
+library(VIBER)
+library(cli)
+library(dplyr)
+library(ggplot2)
+
+outdir = ""
+
+# Auxiliary functions
+
+add_dummy_driver = function(input_table, variant_colname, is_driver_colname) {
+    add_driver = FALSE
+    idx = dplyr::case_when(
+        "cluster" %in% colnames(input_table) ~ which(input_table[["cluster"]] != "Tail")[1],
+        .default = 1
+    )
+    if (!variant_colname %in% colnames(input_table) | !is_driver_colname %in% colnames(input_table)) {
+        input_table = input_table %>% dplyr::mutate(!!is_driver_colname:=FALSE, !!variant_colname:=NA)
+        add_driver = TRUE
+    } else if (all(input_table[[is_driver_colname]]==FALSE)) {
+        add_driver = TRUE
+    }
+    if (add_driver) {
+        input_table[idx, is_driver_colname] = TRUE
+        input_table[idx, variant_colname] = ""
+    }
+    return(input_table)
+}
+
+initialize_ctree_obj_pyclone = function(ctree_input) {
+    driver_cluster = unique(ctree_input[which(ctree_input["is.driver"]==TRUE),c("cluster")])
+    # the CCF table must report CCF values for each cluster and sample
+    # cluster | nMuts | is.driver | is.clonal | sample1 | sample2 | ...
+    CCF_table = ctree_input %>%
+        dplyr::select(sample_id, cluster, nMuts, is.driver, is.clonal, CCF) %>%
+        dplyr::mutate(is.driver=replace(is.driver, is.driver=="", "FALSE")) %>%
+        dplyr::mutate(is.driver=as.logical(is.driver)) %>%
+        dplyr::filter(cluster!="Tail") %>%
+        dplyr::mutate(cluster=as.character(cluster)) %>%
+        dplyr::group_by(cluster) %>%
+        dplyr::mutate(is.driver=any(is.driver)) %>%
+        dplyr::filter(any(CCF>0)) %>%
+        dplyr::ungroup() %>% unique() %>%
+        tidyr::pivot_wider(names_from="sample_id", values_from="CCF", values_fill=0)
+
+    # the driver table must contain patient and variant IDs and report clonality and driver status
+    # patientID | variantID | is.driver | is.clonal | cluster | sample1 | sample2 | ...
+    drivers_table = ctree_input %>%
+        dplyr::filter(cluster %in% CCF_table[["cluster"]]) %>%
+        dplyr::mutate(is.driver=as.logical(is.driver)) %>%
+        dplyr::mutate(cluster=as.character(cluster)) %>%
+        dplyr::select(patientID, sample_id, variantID, cluster, is.driver, is.clonal, CCF) %>%
+        dplyr::filter(is.driver==TRUE) %>%
+        dplyr::mutate(variantID=replace(variantID, is.na(variantID), "")) %>%
+        tidyr::pivot_wider(names_from="sample_id", values_from="CCF", values_fill=0)
+
+    samples = unique(ctree_input[["sample_id"]])  # if multisample, this is a list
+    patient = unique(ctree_input[["patientID"]])
```

Comment:
Unrelated (and probably an issue in the upstream input), but is a bit jarring that you have snake case (`sample_id`), camel case (`patientID`) and dots (`is.driver`).